### PR TITLE
Add support for alternative urn schemes

### PR DESCRIPTION
**Pull request by fabrik42**
Hey,

thank you for this lib. I thought it makes sense to support other URN schemes too as it seems to be an easy modification.

This commit enables other urn schemes than `bitcoin` to be de-/encoded.

Please let me know what you think! 🙂

### CODE REVIEW

**Comment by dcousens** [comment]
@fabrik42 why?
It can be easy to parameterize many things,  but,  this is specifically a **BIP21** implementation?

**Comment by fabrik42** [comment]
Hi @dcousens,
I want to use this lib for the transfer of non-bitcoin related public keys and meta data. BIP21 is a good format for many use cases, not only bitcoin-related and I think some other altcoins like the official Dash app are already using it. It is not uncommon for other altcoins to use some BIP feature libs, if this is possible.

**Comment by dcousens** [comment]
@fabrik42 I think I'd rather that as a fork tbh.  I'd rather this code not become overly complex.

File: index.js [hunk 1]
```diff
@@ -22,14 +23,21 @@ function decode (uri) {
 
 function encode (address, options) {
   options = options || {}
+
+  var scheme = 'bitcoin'
+  if (options.scheme) {
+    scheme = options.scheme
+    delete options.scheme
```
Comment (dcousens) [inline index.js]: FWIW, this would modify the `options` object... bad practice.

**Comment by dcousens** [comment]
Undecided.  For now,  if you want to fix up that nit I pointed out above by not mutating the `options` object :+1:

File: index.js (hunk 1 above)
Comment (fabrik42) [reply index.js]: You are right, I will fix this!

**Comment by fabrik42** [comment]
I changed the interface, to the `encode` function now accepts a third optional parameter, this way it is more streamlined with the `decode` function.

However, I get you point, that you want to keep the lib as simple as possible. If you don't want to add this feature, I can also live with my fork, no offense will be taken. 🙂

**Comment by dcousens** [comment]
@fabrik42 in the end, I think the benefit of having this code under test for those that use it, is better than rejecting it on the principle.  The added complexity is minimal,  and honestly,  I'd rather reduce that complexity by removing the use of regex...

**Comment by afk11** [comment]
Oops, forgot to post with ACK - looks good! :+1:

**Pull request merged**